src/sections/General/Footer/index.js

```diff
@@ -7,8 +7,75 @@ import Button from "../../../reusecore/Button";
 import FooterWrapper from "./footer.style";
 import bubblesElement from "./images/bubbles-element.svg";
 
-const Footer = () => {
+const Footer = ({ location }) => {
   var currentYear = new Date().getFullYear();
+
+  const getUrl = (pathname) => {
+    const indexUrl = ["/", "/blog", "/careers", "/meshery", "/meshmap", "/community", "/handbook", "projects", "/resources", "/learn"];
+    const mdUrlHandle =
+    [
+      {
+        partUrl: "/blog/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/blog"
+      },
+      {
+        partUrl: "/service-mesh-books/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/service-mesh-books"
+      },
+      {
+        partUrl: "/events/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/events"
+      },
+      {
+        partUrl: "/programs/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/programs"
+      },
+      {
+        partUrl: "/careers",
+        gitUrl: "https://github.com/layer5io/layer5/blob/master/src/pages/careers/index.js"
+      },
+      {
+        partUrl: "/careers/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/careers"
+      },
+      {
+        partUrl: "/news/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/news"
+      },
+      {
+        partUrl: "/members/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/members"
+      },
+      {
+        partUrl: "/resource/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/resources"
+      },
+      {
+        partUrl: "/service-mesh-labs/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/service-mesh-labs"
+      },
+      {
+        partUrl: "/service-mesh-workshops/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/src/collections/service-mesh-workshops"
+      },
+      {
+        partUrl: "/learning-paths/",
+        gitUrl: "https://github.com/layer5io/layer5/tree/master/content-learn"
+      },
+    ];
```

Comment:
Considering the gitUrl for every collection starts with `https://github.com/layer5io/layer5/tree/master/src/collections/`, we can save it in one variable and updte the array to have only the partUrl, or even better just update the list from list of objects to simple list having the partUrls as it's value.
The special case of `learning-paths` can be handled individually in the `forEach` loop below.

Reply:
Hi @8rxn,

I agree with Nikhil, the repetition seems to be an opportunity to consolidate things and reduce the amount of code.

Also, there are a lot of ways to check the `pathname` for constructing the github edit url, and I thought [this post](https://daily-dev-tips.com/posts/javascript-startswith-and-multiple-conditions/) had a good option that seemed code efficient.

Hope that helps,
Randy

Reply:
Hm. Understood. I'll  address that change.
While starting work on this I didn't think there'd  be that many links that fall inside collection, later I forgot to change it. 

Will be implementing this.

Reply:
Hi All,

Thought this was an interesting javascript exercise with all the different ways to handle this situation. I have no idea what is the cleanest, most readable, or best performing method, so if you have any ideas, I am happy to learn. 

For fun/sharing, here's a couple of variations using @8rxn's idea with the array of values:

<details><summary>ternary version</summary>

 ```
const pathCheck = pathname.replace(".html", "");
  const mdxCheck = ["/blog/", "/careers/", "/events/", "/integrations/", "/landscape/", "/members/", "/news/", "/programs/", "/projects/", "/resources/", "/service-mesh-books/", "/service-mesh-labs/", "/service-mesh-workshops/"].some((word) => pathCheck.startsWith(word));

  const editUrl = `https://github.com/layer5io/layer5/blob/master/${
    pathCheck.startsWith("/learn/learning-paths/")
      ? `content-learn/${pathCheck.replace("/learn/learning-paths/", "")}`
      : mdxCheck
        ? `src/collections${pathCheck}`
        : pathCheck === "/"
          ? "src/pages/index"
          : `src/pages${pathCheck}`}
    .js`;
```
    
</details>

<details><summary>function version</summary>

 ```
 const getRestOfUrl = (pathname) => {
    //remove ".html" that results in live production build
    if (pathname.endsWith(".html")) {
      pathname = pathname.replace(".html", "");
    }
    // series of tests to structure the return string
    const test = {
      mdx: function(path){
        return ["/blog/", "/careers/", "/events/", "/integrations/", "/landscape/", "/members/", "/news/", "/programs", "/projects/", "/resources/", "/service-mesh-books/", "/service-mesh-labs/", "/service-mesh-workshops/"].some((check) => path.startsWith(check)) && `src/collections${path}`;
      },
      learningPath: function(path) {
        return path.startsWith("/learn/learning-paths/") && `content-learn${pathname.replace("learn/learning-paths/", "")}`;
      }
    };
//returns the first true value from the tests, otherwise returns string
    return test.mdx(pathname) || test.learningPath(pathname) || `pages${pathname}`;
  };

  const editUrl = `https://github.com/layer5io/layer5/blob/master/${getRestOfUrl(pathname)}.js`;

```
    
</details>

cheers,
Randy

Reply:
That's great work @randychilau 

While the ternary version is short, the function version seems much more readable as ternary seems to be having a lot of nesting not great for readability,
One thing to note here is, when the `/pages` directory is used the link is to a ` .js ` file so `/blob/master` , while when the mdx pages are having the link, it redirects to the folder containing all of their mdx files, hence `/tree/master`.

I'll be making tweaks in this code and shamelessly include this in PR :p 😄